components/views/login/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState, FormEvent } from 'react'
+import styles from './Login.module.scss'
+import useForm from '../../../utilities/hooks/useForm'
+import Link from 'next/link'
+
+const Login = () => {
+    const { input, handleChange, handleReset, clearForm } = useForm({
+        email: '',
+        password: ''
+    })
```

Comment:
instead of this being an object defined right here, it should be predefined outside the component and covered with TypeScript types. Then pass the object into user form

```
interface FormProps {
  email: string
  password: string
}

const INITIAL_STATE = {
  email: '',
  password: ''
}
```
Then inside the component:

```
const { input, handleChange, handleReset, clearForm } = useForm<FormProps>useForm(INITIAL_STATE)
```